client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/readclient/CelebornBufferStream.java

```diff
@@ -312,8 +323,19 @@ public void onFailure(Throwable e) {
                 shuffleKey,
                 NettyUtils.getRemoteAddress(client.getChannel()));
             messageConsumer.accept(new TransportableError(streamId, e));
+            if (resultFuture != null) {
+              resultFuture.set(null);
+            }
           }
         });
+
+    if (sync) {
+      try {
+        resultFuture.get();
```

Comment:
We should `openStream` synchronized as `requiredSegmentId` RPC have to be sent after open.